packages/jest-circus/src/utils.ts

```diff
@@ -170,9 +170,15 @@ export const describeBlockHasTests = (
     child => child.type === 'test' || describeBlockHasTests(child),
   );
 
-const _makeTimeoutMessage = (timeout: number, isHook: boolean) =>
+const _makeTimeoutMessage = (
+  timeout: number,
+  isHook: boolean,
+  takesDoneCallback: boolean,
+) =>
   `Exceeded timeout of ${formatTime(timeout)} for a ${
     isHook ? 'hook' : 'test'
+  }${
+    takesDoneCallback && ' while waiting for done to be called'
```

Comment:
Perhaps? Not 100% sure how to make it more clear what "done" is?

```suggestion
    takesDoneCallback && ' while waiting for `done()` to be called'
```